domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs

```diff
@@ -57,6 +57,10 @@ public async Task SeedData(Action<ApplicationDbContext> seeder)
     {
         using var scope = Services.CreateScope();
         var dbContext = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+
+        dbContext.TransferAgreements.RemoveRange(dbContext.TransferAgreements);
```

Comment:
It is to avoid sharing db contexts between integration tests. Before adding this line, the getbysubjectid_list test, would fail, because it was running with the same dbcontext instance as the other tests, thus returning a list of transferagreements containing a higher number of transfer agreements than expected.